Video Optimization: Add missing `crossorigin` attributes

spacedmonkey opened the pull request:
## Context

<!-- What do we want to achieve with this PR? Why did we write this code? -->

## Summary

This is a follow-up to #6875 .

With these new headers, 3P media are no longer loaded due to missing crossorigin attributes.

## Relevant Technical Choices

I have also improved poster images, by making sure, loading=lazy, alt and onload are loaded on poster images as well. 

This PR will likely fix another issue, where images from media library are served from another url, like if assets are loaded from s3.

## To-do

<!-- A list of things that need to be addressed in this PR or follow-up changes. -->

## User-facing changes

<!--
Please describe your changes.
Include before/after screenshots or a short video.
-->

## Testing Instructions

<!--
How can the changes in this PR be verified?
Please provide step-by-step instructions how to reproduce the issue, if applicable.
Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->

### QA

<!--
Not all changes require manual QA.
-->

<!-- ignore-task-list-start -->
- [ ] This is a non-user-facing change and requires no QA
<!-- ignore-task-list-end -->

This PR can be tested by following these steps:

1. Enable video optimization experiment
2. Browsing and using 3P Media (videos, images, GIFs) should work without issues or console errors  
    **Note:** there can be some console errors for GIFs due to caching, but most of them should load fine already.
3. Toggling "show images" in Page templates tab should not cause console errors

### UAT

<!--
Sometimes the testing instructions for UAT can differ from the ones for QA.
-->

<!-- ignore-task-list-start -->
- [ ] UAT should use the same steps as above.
<!-- ignore-task-list-end -->

<!--
If the above checkbox has not been checked, write down all steps necessary for user acceptance testing take to test this PR.
-->
This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

<!-- Examples: new APIs, changes to KSES, etc.  -->

### Does this PR change what data or activity we track or use?

<!-- Examples: changes to telemetry, new third-party APIs -->

### Does this PR have a legal-related impact?

<!-- Examples: new images with unknown sources, new production dependencies with incompatible licenses -->

## Checklist

<!-- Check these after PR creation -->

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

<!--
Please reference the issue(s) this PR addresses.
No URLs, just the issue numbers.
Use "Fixes #123" if it fixes an issue.
Example:

Fixes #123
Partially addresses #456
See #789
-->

Fixes #7016

spacedmonkey committed:
Add cross origin to images in editor.

spacedmonkey committed:
Revert change.

spacedmonkey committed:
Revert change.

swissspidy commented on assets/src/edit-story/components/library/panes/media/common/innerElement.js:
```diff
@@ -134,20 +137,31 @@ function InnerElement({
+
+  const commonImageProps = {
     ...commonProps,
+    onLoad: makeMediaVisible,
```
Why was `onLoad` moved from `imageProps` to the common props here? 🤔

spacedmonkey replied:
> I have also improved poster images, by making sure, loading=lazy, alt and onload are loaded on poster images as well.

spacedmonkey replied:
This just makes the image props between poster image and image the same.

swissspidy replied:
But I don't think the poster needs an `onLoad` callback?

spacedmonkey replied:
The onload just fades in the image. Poster image are image tags like an image, they should fail in the same way a image does.

spacedmonkey replied:
Also It already has a onload- https://github.com/google/web-stories-wp/pull/7086/files#diff-156fd9c14d28faad8ca9b94fd115f288b4a1511b74ac5b3c2bdd5db277f0ab91L193 This isnt a change.

spacedmonkey committed:
Merge branch 'main' into fix/cross-origin-3p

spacedmonkey committed:
Add crossOrigin to all image and video tags.

spacedmonkey committed:
Revert change.

spacedmonkey committed:
Remove siteUrl.

spacedmonkey committed:
Revert unneeded changes.

spacedmonkey commented on assets/src/web-stories-block/block/block-types/story-embed-block/embedControls.js:
```diff
@@ -128,7 +128,7 @@ const EmbedControls = (props) => {
                   {__('Poster image', 'web-stories')}
                 </BaseControl.VisualLabel>
-                {hasPoster && <img src={poster} alt="" />}
+                {hasPoster && <img src={poster} alt="" crossOrigin="anonymous" />}
```
```suggestion
                {hasPoster && <img src={poster} alt="" />}
```

spacedmonkey committed:
Update assets/src/web-stories-block/block/block-types/story-embed-block/embedControls.js

spacedmonkey committed:
Merge branch 'main' into fix/cross-origin-3p

swissspidy committed:
Merge branch 'main' into fix/cross-origin-3p

swissspidy committed:
Avoid double slashes for template URLs

`imageBaseUrl` (`cdnURL`) will always have a trailing slash.

This change prevents unnecessary 301 redirects on the CDN.

swissspidy committed:
Undo unrelated change

swissspidy committed:
Lint fix

spacedmonkey committed:
Merge branch 'fix/cross-origin-3p' of github.com:google/web-stories-wp into fix/cross-origin-3p

spacedmonkey committed:
Remove crossOrigin on quick tips.

swissspidy committed:
Merge branch 'main' into fix/cross-origin-3p

miina commented on assets/src/edit-story/components/library/panes/media/common/innerElement.js:
```diff
@@ -136,20 +136,28 @@ function InnerElement({
+  };
+
+  const cloneProps = {
+    ...commonImageProps,
```
> This just makes the image props between poster image and image the same.

@spacedmonkey `cloneProps` shouldn't really need `onLoad`, what was the reason for adding it there as well? Clone is the visible clone image while dragging media from the library onto a page, it only becomes visible once starting dragging an image/video. Thoughts?

swissspidy replied:
Definitely looks like a bug. Reverted in d7b236efb7ba2f7515509d9e2afe11ad791b6882

swissspidy committed:
Merge branch 'main' into fix/cross-origin-3p

swissspidy committed:
Add back `crossOrigin` attribute for quick tips videos

Reverts 277522b2f993bd9d316cc9f15fd200ef6453d351

swissspidy committed:
Set `crossOrigin` attribute in `preloadImage`

swissspidy committed:
Don’t set `onLoad` on `cloneProps`

swissspidy committed:
Merge branch 'main' into fix/cross-origin-3p

swissspidy committed:
Revert more changes in `InnerElement`

swissspidy committed:
3P Media: don’t show video duration for GIFs

swissspidy committed:
Fix duplicate `aria-label`

swissspidy committed:
Merge branch 'main' into fix/cross-origin-3p

swissspidy committed:
Set poster ID for 3p media

Prevents accidental poster generation for 3p media in places like `useInsertElement` that look for the poster ID to determine whether one needs to be generated or not.

swissspidy committed:
Merge branch 'main' into fix/cross-origin-3p

swissspidy committed:
Update tests
